app/api/users.py

```diff
@@ -14,17 +14,31 @@ def get_user(username):
     abort(404)
 
 
-@api_bp.route('/edit_profile/<user_id>')
+@api_bp.route('/update_profile/', methods=["GET", "POST"])
 @token_auth.login_required
-def edit_profile():
-    is_sitter = request.get_json()['is_sitter']
-    collection = Sitter if is_sitter else Owner
-    token = request.get_json()['token']
-    u = collection.check_token(token)
-    return jsonify(u.to_dict(include_email=True))
+def update_profile():
+   is_sitter = request.get_json()['is_sitter']
+   collection = Sitter if is_sitter else Owner
+   print(request.get_json()['profile_data'])
+   profile_data = request.get_json()['profile_data']
+   birthdate = profile_data['birthdate']
+   gender = profile_data['gender']
+   about_me = profile_data['about_me']
+   location = profile_data['location']
+   charge = profile_data['rate']
+
+   token = request.get_json()['token']
+   u = collection.check_token(token)
```

Comment:
you should haven't to check token here again

Reply:
yes once we include the fetching of the user inside decorator we will remove this,
here it is just for the sake of making it work for the time being.